build/teamcity/internal/cockroach/release/publish/sign_macos_release.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+set -xeuo pipefail
+
+KEYCHAIN_NAME=signing
+KEYCHAIN_PROFILE=notarization
+curr_dir=$(pwd)
+
+remove_files_on_exit() {
+  rm -f "$curr_dir/.google-credentials.json"
+  security lock-keychain "${KEYCHAIN_NAME}"
+}
+trap remove_files_on_exit EXIT
+
+# By default, set dry-run variables
+google_credentials="$GCS_CREDENTIALS_DEV"
+s3_bucket="cockroach-builds-test"
+gcs_bucket="cockroach-release-artifacts-dryrun"
+
+# override dev defaults with production values
+if [[ -z "${DRY_RUN}" ]] ; then
+  echo "Setting production values"
+  google_credentials="$GCS_CREDENTIALS_PROD"
+  s3_bucket="binaries.cockroachdb.com"
+  gcs_bucket="cockroach-release-artifacts-prod"
+fi
+
+echo "$google_credentials" > "$curr_dir/.google-credentials.json"
+export GOOGLE_APPLICATION_CREDENTIALS="$curr_dir/.google-credentials.json"
+
+# install bazelisk
+mkdir -p bazelisk-bin
+curl -fsSL https://github.com/bazelbuild/bazelisk/releases/download/v1.13.2/bazelisk-darwin-arm64 > bazelisk-bin/bazel
+shasum --algorithm 256 --check - <<EOF
+c84fb9ae7409b19aee847af4767639ab86fc3ce110f961107ff278879b019e38  bazelisk-bin/bazel
+EOF
+chmod 755 bazelisk-bin/bazel
+export PATH=$PWD/bazelisk-bin:$PATH
```

Comment:
I'm curious, why install bazelisk on every run instead of installing it on the agent image?

Reply:
This agent is a long living one. Installing bazelisk on it would help now, but if we want to use a different version, this would make that installation obsolete.